tests/hostpath_quota_test.go

```diff
@@ -0,0 +1,463 @@
+			By("building a PVC with StorageClass " + scName)
+			pvcObj, err = BuildPersistentVolumeClaim(pvcName, scName, capacity, accessModes)
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building PVC {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+
+			By("creating above PVC with StorageClass " + scName)
+			createdPvc, err = ops.PVCClient.WithNamespace(namespaceObj.Name).Create(context.TODO(), pvcObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating PVC {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+		})
+	})
+
+	When("pod is created with pvc "+pvcName, func() {
+		It("should be in pending state and so do pvc", func() {
+			By("building a pod with busybox image")
+			podObj, err = BuildPod(podName, pvcName, labelselector)
+			Expect(err).ShouldNot(
+				HaveOccurred(),
+				"while building pod {%s} in namespace {%s}",
+				podName,
+				namespaceObj.Name,
+			)
+
+			By("creating above pod with busybox image")
+			createdPod, err := ops.PodClient.WithNamespace(namespaceObj.Name).
+				Create(context.TODO(), podObj)
+			Expect(err).To(
+				BeNil(),
+				"while creating pod {%s} in namespace {%s}",
+				podName,
+				namespaceObj.Name,
+			)
+
+			By("verifying pod status as pending")
+			checkPhase := ops.CheckPodStatusEventually(createdPod, corev1.PodPending)
+			Expect(checkPhase).To(Equal(true), "while verifying pod pending status")
+
+			By("verifying the pvc phase as pending")
+			Expect(createdPvc.Status.Phase).To(Equal(corev1.ClaimPending), "while verifying the pvc pending state")
+		})
+	})
+
+	When("Pod consuming pvc "+pvcName+" is deleted", func() {
+		It("should delete the pod", func() {
+			By("deleting above pod")
+			err = ops.PodClient.WithNamespace(namespaceObj.Name).Delete(context.TODO(), podName, &metav1.DeleteOptions{})
+			Expect(err).To(
+				BeNil(),
+				"while deleting pod {%s} in namespace {%s}",
+				podName,
+				namespaceObj.Name,
+			)
+
+			By("verifying pod count as 0")
+			podCount := ops.GetPodCountEventually(namespaceObj.Name, label, nil, 0)
+			Expect(podCount).To(Equal(0), "while verifying pod count")
+		})
+	})
+
+	When("pvc with storageclass "+scName+" is deleted", func() {
+		It("should delete the pvc", func() {
+			By("deleting the PVC with StorageClass " + scName)
+			pvName := ops.GetPVNameFromPVCName(pvcName)
+			Expect(pvName).To(
+				BeEmpty(),
+				"while getting Spec.VolumeName from PVC {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+
+			By("deleting above PVC")
+			err = ops.PVCClient.Delete(context.TODO(), pvcName, &metav1.DeleteOptions{})
+			Expect(err).To(
+				BeNil(),
+				"while deleting pvc {%s} in namespace {%s}",
+				pvcName,
+				namespaceObj.Name,
+			)
+
+			By("verifying PVC is deleted")
+			status := ops.IsPVCDeletedEventually(pvcName, namespaceObj.Name)
+			Expect(status).To(
+				BeTrue(),
+				"when checking status of deleted PVC {%s}",
+				pvcName,
+			)
+		})
+	})
```

Comment:
We also need to delete the StorageClass created in step1

Reply:
That is handled in AfterSuite block.